test/framework/src/main/java/org/elasticsearch/test/rest/yaml/Stash.java

```diff
@@ -105,7 +105,7 @@ public Object getValue(String key) throws IOException {
     }
 
     private Object unstash(String key) throws IOException {
-        Object stashedValue = stashObjectPath.evaluate(key);
+        Object stashedValue = stashObjectPath.evaluate(key, this);
```

Comment:
I will extract this into a separate PR with a corresponding unit test in `StashTests`.

Reply:
This is change is needed, because specifying a path with multiple variables (like `$body.$backing_index.data_stream`) isn't currently possible. Because the stash
isn't passed around when doing subsequent variable evaluation.

Also just specifying `$backing_index.data_stream` also doesn't work, because then 
the yaml infrastructure wants to resolve data_stream from the key stashed under `backing_index`,
which just contains the string of the backing index. Therefor first specifying `$body` is required, because
then the response body of the previous request is available for evaluation. It looks like this is only
a problem if the first element of a path is being substituted with a variable.